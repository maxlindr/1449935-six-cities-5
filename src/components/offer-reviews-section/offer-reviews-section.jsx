import React from 'react';
import PropTypes from 'prop-types';
import {userPropTypes, reviewPropTypes} from '../../prop-types';
import withReviewFormController from '../../hocs/with-review-form-controller/with-review-form-controller';
import ReviewForm from '../review-form/review-form';
import OfferReview from '../offer-review/offer-review';

const ReviewFormComponent = withReviewFormController(ReviewForm);

const OfferReviewsSection = (props) => {
  const {user, reviews} = props;

  return (
    <section className="property__reviews reviews">
      <h2 className="reviews__title">Reviews Â· <span className="reviews__amount">{reviews.length}</span></h2>

      <ul className="reviews__list">
        {reviews.map((review) => <OfferReview key={review.id} review={review}/>)}
      </ul>

      {user ? <ReviewFormComponent /> : null}
    </section>
  );
};

OfferReviewsSection.propTypes = {
  reviews: PropTypes.arrayOf(reviewPropTypes),
  user: userPropTypes,
};

export default OfferReviewsSection;
