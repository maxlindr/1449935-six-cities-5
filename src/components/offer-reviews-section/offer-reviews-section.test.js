import React from 'react';
import renderer from 'react-test-renderer';
import reviewMocks from '../../../__mocks__/review-mocks';
import {OfferReviewsSection} from './offer-reviews-section';

test(`OfferReviewsSection должен корректно отображаться`, () => {
  const tree = renderer.create(
      <OfferReviewsSection
        reviews={reviewMocks}
        offerId={`1`}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
