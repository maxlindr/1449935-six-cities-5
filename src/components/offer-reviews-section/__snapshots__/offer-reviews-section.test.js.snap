// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferReviewsSection должен корректно отображаться 1`] = `
<section
  className="property__reviews reviews"
>
  <h2
    className="reviews__title"
  >
    Reviews · 
    <span
      className="reviews__amount"
    >
      4
    </span>
  </h2>
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height={54}
            src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/2.jpg"
            width={54}
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Jack
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          This villa is perfect in every way: the view on mountains and waterfalls, the hot tub and the villa itself. The evening here became a great continuation of our journeys over country.
        </p>
        <time
          className="reviews__time"
          dateTime="2020-10-10"
        >
          October 2020
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height={54}
            src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/9.jpg"
            width={54}
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Sophie
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "40%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          I stayed here for one night and it was an unpleasant experience.
        </p>
        <time
          className="reviews__time"
          dateTime="2020-10-10"
        >
          October 2020
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height={54}
            src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/10.jpg"
            width={54}
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Christina
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "80%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          I stayed here for one night and it was an unpleasant experience.
        </p>
        <time
          className="reviews__time"
          dateTime="2020-10-10"
        >
          October 2020
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height={54}
            src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/7.jpg"
            width={54}
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Oliver.conner
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          oh it was excellent oh it was excellent oh it was excellent
        </p>
        <time
          className="reviews__time"
          dateTime="2020-11-10"
        >
          November 2020
        </time>
      </div>
    </li>
  </ul>
</section>
`;
